src/SampleLibraryUI/Examples/ButtonCustomNodeModel.cs

```diff
@@ -0,0 +1,336 @@
+
+            // Using the AstFactory class, we can build AstNode objects
+            // that assign doubles, assign function calls, build expression lists, etc.
+            return new[]
+            {
+                // In these assignments, GetAstIdentifierForOutputIndex finds 
+                // the unique identifier which represents an output on this node
+                // and 'assigns' that variable the expression that you create.
+
+                AstFactory.BuildAssignment(GetAstIdentifierForOutputIndex(0), buttonFuncNode),
+                    AstFactory.BuildAssignment(
+                        AstFactory.BuildIdentifier(AstIdentifierBase + "_dummy"),
+                        VMDataBridge.DataBridge.GenerateBridgeDataAst(GUID.ToString(), AstFactory.BuildExprList(inputAstNodes))),
+
+                AstFactory.BuildAssignment(GetAstIdentifierForOutputIndex(1), windowFuncNode),
+                    AstFactory.BuildAssignment(
+                        AstFactory.BuildIdentifier(AstIdentifierBase + "_dummy"),
+                        VMDataBridge.DataBridge.GenerateBridgeDataAst(GUID.ToString(), AstFactory.BuildExprList(inputAstNodes)))
+            };
+        }
+
+        #endregion
+
+        #region command methods
+        private bool CanShowMessage(object obj)
+        {
+            return true;
+        }
+
+        private void ShowMessage(object obj)
+        {
+            MessageBox.Show(WindowText);
+        }
+
+        #endregion
+    }
+
+    /// <summary>
+    ///     View customizer for CustomNodeModel Node Model.
+    /// </summary>
+    public class ButtonCustomNodeModelNodeViewCustomization : INodeViewCustomization<ButtonCustomNodeModel>
+    {
+        /// <summary>
+        /// At run-time, this method is called during the node 
+        /// creation. Here you can create custom UI elements and
+        /// add them to the node view, but we recommend designing
+        /// your UI declaratively using xaml, and binding it to
+        /// properties on this node as the DataContext.
+        /// </summary>
+        /// <param name="model">The NodeModel representing the node's core logic.</param>
+        /// <param name="nodeView">The NodeView representing the node in the graph.</param>
+        public void CustomizeView(ButtonCustomNodeModel model, NodeView nodeView)
+        {
+            // The view variable is a reference to the node's view.
+            // In the middle of the node is a grid called the InputGrid.
+            // We reccommend putting your custom UI in this grid, as it has
+            // been designed for this purpose.
+
+            // Create an instance of our custom UI class (defined in xaml),
+            // and put it into the input grid.
+            var buttonControl = new ButtonControl();
+            nodeView.inputGrid.Children.Add(buttonControl);
+
+            // Set the data context for our control to be this class.
```

Comment:
this comment is wrong - you are setting the datacontext to the nodeModel... not the custmization right?